upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -700,6 +716,39 @@ func (c *ApplyClusterCmd) Run() error {
 
 	tf.AddTo(l.TemplateFunctions)
 
+	if cluster.Spec.Assets != nil && cluster.Spec.Assets.ContainerRegistry != nil {
+		kubernetesComponents := []string{"kube-proxy", "kube-apiserver", "kube-controller-manager", "kube-scheduler"}
```

Comment:
But because we are not passing around the same assets builder we are loosing the information.  I am rebuilding it here.  Any recommendations?  

Yah we need to cleanup api server, and git rid of the non go model.